state/backups/db_restore_test.go

```diff
@@ -134,6 +147,10 @@ func (s *mongoRestoreSuite) TestRestoreDatabase32(c *gc.C) {
 			bson.DocElem{Name: "grantRolesToUser", Value: "admin"},
 			bson.DocElem{Name: "roles", Value: []string{"oploger"}}}}
 	c.Assert(mgoSession.cmd, gc.DeepEquals, mgoSessionCmd)
+
+	// Run a 2nd time, lp:1740969
+	err = restorer.Restore("fakePath", nil)
+	c.Assert(err, jc.ErrorIsNil)
```

Comment:
Besides checking nil error, is there anything else we can check to see that stuff has run as expected.
Perhaps a subset of what was checked for the first run. Or if we delete the databases and restore into an empty instance, just re-run all the checks. You could package stuff into a method and do something like:

func assertRestore() {
}

func TestRestore() {
  assertRestore()
}

func TestRestoreIdempotent() {
  assertRestore()
  assertRestore()
}

Reply:
took the assertRestore option.  i have a note on the other.